### DISCUSSION TITLE
Get branch commits returns only head commit

### DISCUSSION THREAD

**Discussion by altonotch** [community/community#24785] [Apps, API and Webhooks]
<p>Hi guys,</p>
<p>I’m looking for a way to get last N commits for a branch.<br>
I am trying to follow this: <a href="https://docs.github.com/en/rest/reference/repos#commits" class="inline-onebox" rel="noopener nofollow ugc">Repositories - GitHub Docs</a>.<br>
Calling <code>/repos/{owner}/{repo}/commits/{branch_name}</code>  returns only the head commit for the branch.<br>
Optimally I would like to detect when this branch checked out from master and get all commits in it.<br>
Sub-optimally will be ok to get last N commits for the branch.</p>
<p>Is there a way to do this via github rest api?</p>
<p>Any help with this will be greatly appreciated :pray:</p>

**Comment by altonotch** [comment, marked as answer]
<p>Figured it out. It’s a linked list, follow <code>parents</code> <code>url</code> and keep traversing.</p>

**Comment by aadeethyeah** [comment]
is there a way to get only feature branch's commits 
because when I am using this github api **(/repos/{owner}/{repo}/commits)** with query param as **sha=branch name**
 it gives feature branches commit and goes back to the commits present from start of the repo.
like it is bringing even the commits from where we created the feature branch.

<img width="669" alt="commit" src="https://user-images.githubusercontent.com/104893727/211724821-0631a22c-6516-47c8-bb8a-bccdae493aa2.PNG">
